src/common/intercom.js

```diff
@@ -36,7 +36,33 @@ function intercomShow() {
 }
 
 function trackIntercomEvent(event, metadata) {
-    console.log("track event");
+    console.log("track event: " + event + " " + JSON.stringify(metadata));
+    if (window.Intercom) {
+        if (metadata) {
+            window.Intercom("trackEvent", event, metadata);
+        } else {
+            window.Intercom("trackEvent", event);
+        }
+    }
 }
 
-export { intercomLogin, intercomLogout, intercomShow, trackIntercomEvent };
+const IntercomEvents = {
+    VIEWED_FOLDER: "viewed folder",
+    VIEWED_FILE: "viewed file",
+    VIEWED_APPS: "viewed apps",
+    VIEWED_ANALYSES: "viewed analyses",
+    VIEWED_TOOLS: "viewed tools",
+    VIEWED_NOTIFICATIONS: "viewed notifications",
+    LAUNCHED_JOB: "launched job",
+    VIEWED_HELP: "viewed help",
+    URL_IMPORT_SUBMITTED: "url import submitted",
+    FILE_UPLOADED: "file uploaded",
+};
```

Comment:
This is not an exhaustive list of events. Suggestions welcome on what other events when tracked in intercom will be useful.

Reply:
We have teams now, so we could add that.  
I wonder how difficult it would be to track when users are seeing the ErrorHandler or some of the other error components?  Not sure how straightforward that is since it looks like the pattern is to add these events after an API call.  One of the easier things might be to add a tracker when `showErrorAnnouncer` is called [here](https://github.com/cyverse-de/sonora/blob/master/src/components/utils/error/withErrorAnnouncer.js#L46) and we could add the `text` and `error` data perhaps.

Reply:
I'm wondering why we should only track events when the API returns a success response? Would it better to track the event whenever the user selects a related action in the UI, so we can see what they did no matter if the API returns a success or error response?

For example, maybe we can track the `VIEWED_APPS` event whenever the Apps icon is selected from the main nav bar, or whenever they change the listing filter; and the `VIEWED_FOLDER` event could trigger when they select a folder name from the data listing, or when selecting the "View Outputs" action in the analyses listing. In fact, those could be separate events, whereas we can only have 1 right now in the success callback which can't tell how the user got there in the first place 🤔

Reply:
I'd also like to suggest events for viewing metadata, templates, and submitting a DOI request.

Reply:
I agree We can track when Apps Icon is clicked or when `View Output Folder` is clicked. But this could be lead to flood of events 🤔 ?  So we need to selective IMO. Also I would do this in addition to tracking  events on `success` or `error` response and not replace it. For example: When a job is launched, the response contains some useful information like `analysis id`.

Reply:
I think we should also track search related events.

Reply:
I think I have added some events to track based on the feedback. Tracking things that involve nextjs `Link` (e.g: Go to output folder) seems to challenging as we are allowed to write custom logic `onClick`.